dumb/src/it/readpsm.c

```diff
@@ -1246,7 +1246,8 @@ int pattcmp( const unsigned char * a, const unsigned char * b, size_t l )
 	}
 
 	if ( i < j ) return -1;
-	else if ( j > i ) return 1;
+	/* It's same as above. Typo? */
```

Comment:
Did you even read what this is supposed to be doing? This is supposed to be comparing two strings of arbitrary length, which may or may not end with numbers, and I make no assumptions about the length of the part of the string before the number appears. The longer of the two prefixes is supposed to win out before the numbers are compared.